src/callbacks/get_callbacks.jl

```diff
@@ -354,5 +354,21 @@ function get_callbacks(config, sim_info, atmos, params, Y, p)
             (callbacks..., call_every_dt(rrtmgp_model_callback!, dt_rad))
     end
 
+    if atmos.non_orographic_gravity_wave isa NonOrographicGravityWave
+        dt_nogw =
+            dt isa ITime ? ITime(time_to_seconds(parsed_args["dt_nogw"])) :
+            FT(time_to_seconds(parsed_args["dt_nogw"]))
+        dt_nogw, _, _, _ = promote(dt_nogw, t_start, dt, sim_info.t_end)
+        # We use Millisecond to support fractional seconds, eg. 0.1
+        dt_nogw_ms = Dates.Millisecond(1_000 * float(dt_nogw))
+        if parsed_args["dt_save_state_to_disk"] != "Inf" &&
+           !CA.isdivisible(dt_save_state_to_disk_dates, dt_nogw_ms)
+            @warn "Non-orographic gravity wave period ($(dt_nogw_ms)) is not an even divisor of the checkpoint frequency ($dt_save_state_to_disk_dates)"
+            @warn "This simulation will not be reproducible when restarted"
+        end
+
+        callbacks = (callbacks..., call_every_dt(nogw_model_callback!, dt_nogw))
```

Comment:
Note that this callback modifies the state, so the order of callback matters.

I don't know whether this is important or not.

Reply:
I think this callback only modifies the tendencies, not the state?

Reply:
Oh, I was misled by how the function is called: 
```
non_orographic_gravity_wave_compute_tendency!(Y, p)
```

But it makes sense, we are already doing this in other places!